src/code/ContainerRegistryServerAPICalls.cs

```diff
@@ -1115,23 +1115,27 @@ private static Collection<KeyValuePair<string, string>> GetDefaultHeaders(string
         #endregion
 
         #region Publish Methods
-
+        
         /// <summary>
         /// Helper method that publishes a package to the container registry.
         /// This gets called from Publish-PSResource.
         /// </summary>
-        internal bool PushNupkgContainerRegistry(string psd1OrPs1File,
+        internal bool PushNupkgContainerRegistry(
             string outputNupkgDir,
             string packageName,
             string modulePrefix,
             NuGetVersion packageVersion,
             ResourceType resourceType,
             Hashtable parsedMetadataHash,
             Hashtable dependencies,
+            bool isNupkgPathSpecified,
+            string originalNupkgPath,
             out ErrorRecord errRecord)
         {
             _cmdletPassedIn.WriteDebug("In ContainerRegistryServerAPICalls::PushNupkgContainerRegistry()");
-            string fullNupkgFile = System.IO.Path.Combine(outputNupkgDir, packageName + "." + packageVersion.ToNormalizedString() + ".nupkg");
+
+            // if isNupkgPathSpecified, then we need to publish the original .nupkg file, as it may be signed
+            string fullNupkgFile = !isNupkgPathSpecified ? System.IO.Path.Combine(outputNupkgDir, packageName + "." + packageVersion.ToNormalizedString() + ".nupkg") : originalNupkgPath;
```

Comment:
Maybe flip the condition, might make it easier to understand

```suggestion
            string fullNupkgFile = isNupkgPathSpecified ? originalNupkgPath :           System.IO.Path.Combine(outputNupkgDir, packageName + "." + packageVersion.ToNormalizedString() + ".nupkg");
```